### PR TITLE
[Inductor] add inductor config: masked_vec

### PR DESCRIPTION
**Pull request by jiayisunx**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #134566

This PR adds inductor config: masked_vec to control enable/disable masked vectorization for the tail_loop, and enable by default.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/config.py [hunk 1]
```diff
@@ -114,6 +114,9 @@ def autotune_remote_cache_default() -> Optional[bool]:
 # set to True to enable the back-to-back GEMM pass
 b2b_gemm_pass = False
 
+# Whether to enable masked vectorization for the tail_loop
+masked_vec = True
+
```
Comment (jgong5) [inline torch/_inductor/config.py:119]: Please move it to config.cpp and name it `config.cpp.enable_loop_tail_vec`.

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: torch/_inductor/config.py (hunk 1 above)
Comment (jiayisunx) [reply torch/_inductor/config.py]: Done, thanks for your advice!

**Review by jansel — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

**Pull request closed without merging**